public/pages/workflows/new_workflow/new_workflow.tsx

```diff
@@ -29,9 +29,85 @@ import {
 import { enrichPresetWorkflowWithUiMetadata } from './utils';
 import { getDataSourceId, isDataSourceReady } from '../../../utils';
 import { getDataSourceEnabled } from '../../../services';
+import semver from 'semver';
+import { DataSourceAttributes } from '../../../../../../src/plugins/data_source/common/data_sources';
+import { getSavedObjectsClient } from '../../../../public/services';
+import {
+  WORKFLOW_TYPE,
+  MIN_SUPPORTED_VERSION,
+  MINIMUM_FULL_SUPPORTED_VERSION,
+} from '../../../../common/constants';
 
 interface NewWorkflowProps {}
 
+export const getEffectiveVersion = async (
+  dataSourceId: string | undefined
+): Promise<string> => {
+  try {
+    if (dataSourceId === undefined) {
+      throw new Error('Data source is required');
+    }
+
+    const dataSource = await getSavedObjectsClient().get<DataSourceAttributes>(
+      'data-source',
+      dataSourceId
+    );
+    const version =
+      dataSource?.attributes?.dataSourceVersion || MIN_SUPPORTED_VERSION;
+    return version;
+  } catch (error) {
+    console.error('Error getting version:', error);
+    return MIN_SUPPORTED_VERSION;
+  }
+};
+
+const filterPresetsByVersion = async (
+  workflows: WorkflowTemplate[],
+  dataSourceId: string | undefined
+): Promise<WorkflowTemplate[]> => {
+  // if MDS is disabled, skip the version check and assume it is version 2.19+
+  const dataSourceEnabled = getDataSourceEnabled().enabled;
+  if (!dataSourceEnabled) {
+    return workflows;
+  }
+
+  if (!dataSourceId) {
+    return [];
+  }
+
+  const allowedPresetsFor217 = [
+    WORKFLOW_TYPE.SEMANTIC_SEARCH,
+    WORKFLOW_TYPE.MULTIMODAL_SEARCH,
+    WORKFLOW_TYPE.HYBRID_SEARCH,
+  ];
+
+  try {
+    const version = await getEffectiveVersion(dataSourceId);
+
+    if (semver.lt(version, MIN_SUPPORTED_VERSION)) {
+      return [];
+    }
+
+    if (
+      semver.gte(version, MIN_SUPPORTED_VERSION) &&
+      semver.lt(version, MINIMUM_FULL_SUPPORTED_VERSION)
+    ) {
+      return workflows.filter((workflow) => {
+        const workflowType =
+          workflow.ui_metadata?.type ?? WORKFLOW_TYPE.UNKNOWN;
+        return allowedPresetsFor217.includes(workflowType as WORKFLOW_TYPE);
+      });
+    }
+
+    return workflows;
+  } catch (error) {
+    return workflows.filter((workflow) => {
+      const workflowType = workflow.ui_metadata?.type ?? WORKFLOW_TYPE.UNKNOWN;
+      return allowedPresetsFor217.includes(workflowType as WORKFLOW_TYPE);
+    });
+  }
+};
```

Comment:
What could throw errors here / why is a try/catch needed? I don't see anywhere where some unexpected behavior could arise. And it seems because of the try/catch, there's duplicate code on lines 95-99 and 104-107.

Reply:
make sense. Remove the try-catch block in this filter function since it is already done in getVersion()